Core/include/TauIdResults.h

```diff
@@ -70,18 +70,18 @@ struct TauIdDescriptor {
     std::string ToStringRaw() const;
 
     template<typename Tuple, typename Tau>
-    void FillTuple(Tuple& tuple, const Tau& tau, float default_value, const std::string& prefix = "",
+    void FillTuple(Tuple& tuple, const Tau* tau, float default_value, const std::string& prefix = "",
                    const std::string& raw_suffix = "raw") const
     {
         const std::string disc_name = ::analysis::ToString(discriminator);
         if(has_raw){
-            float_t value =  tau.tauID(raw_name);
+            float_t value =  tau ? tau->tauID(raw_name) : default_value;
             tuple.template get<std::vector<float_t>>(prefix + disc_name + raw_suffix).push_back(value);
         }
         if(!working_points.empty()) {
             DiscriminatorIdResults id_results;
             for(const auto& wp_entry : working_points) {
-                const bool result = tau.tauID(wp_entry.second) > 0.5;
+                const bool result = tau ? tau->tauID(wp_entry.second) > 0.5 : default_value;
```

Comment:
for bool default_value is false, so `const bool result = tau && tau->tauID(wp_entry.second) > 0.5`